### PR TITLE
Fix inconsistent behavior caused by `select_sequential_consistency` for optimized trivial count query and system.tables

### PR DESCRIPTION
**Pull request by haoch**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Bug Fix

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):

Fix inconsistent behavior caused by `select_sequential_consistency` for optimized trivial count query and system.tables.

Detailed description / Documentation draft:

We got the following error even with `select_sequential_consistency=0` (disabled), it's very confusing for clickhouse users. 

```
clickhouse-client :) SET select_sequential_consistency = 0

clickhouse-client :) SELECT count(*) from TABLE
Received exception from server (version 20.7.2):
Code: 289. DB::Exception: Received from localhost:9000. DB::Exception: Replica doesn't have part 20200928-0_23211_23211_0 which was successfully written to quorum of other replicas. Send query to another replica or disable 'select_sequential_consistency' setting.: While executing Tables.
```
The root cause is when `optimize_trivial_count_query=1`, trivial count query will call directly`StorageReplicatedMergeTree.totalRows() --> 
 getMaxAddedBlocks()` (with sequential consistency check) at: [StorageReplicatedMergeTree.cpp#getMaxAddedBlocks](https://github.com/ClickHouse/ClickHouse/blob/master/src/Storages/StorageReplicatedMergeTree.cpp#L3503-L3508)

The error will also happen when query `select total_rows, total_bytes system.tables`.

By adding documentation, you'll allow users to try your new feature immediately, not when someone else will have time to document it later. Documentation is necessary for all features that affect user experience in any way. You can add brief documentation draft above, or add documentation right into your patch as Markdown files in docs folder.

If you are doing this for the first time, it's recommended to read the lightweight Contributing to ClickHouse Documentation guide first.

The Information about CI checks: https://clickhouse.tech/docs/en/development/continuous-integration/

### CODE REVIEW

**Comment by haoch** [comment]
This PR is to resolve #16108

File: src/Storages/StorageReplicatedMergeTree.cpp [hunk 1]
```diff
@@ -3711,10 +3723,10 @@ std::optional<UInt64> StorageReplicatedMergeTree::totalRowsByPartitionPredicate(
     return res;
```
Comment (amosbird) [inline src/Storages/StorageReplicatedMergeTree.cpp:3723]: We also need to add context here.

Comment (haoch) [reply src/Storages/StorageReplicatedMergeTree.cpp:3723]: The context already exists in the method without change.

Comment (amosbird) [reply src/Storages/StorageReplicatedMergeTree.cpp:3723]: I mean the `context` needs to be passed into `foreachCommittedParts` . And there are other build errors https://clickhouse-test-reports.s3.yandex.net/16309/feaab260d8e591371febab76b8e26c726111d8aa/fast_test/runlog.out.log

Comment (haoch) [reply src/Storages/StorageReplicatedMergeTree.cpp:3723]: Ok, I have fixed all build errors now.

File: tests/integration/test_reload_zookeeper/test.py [hunk 2]
```diff
@@ -8,7 +8,7 @@
 
 
 cluster = ClickHouseCluster(__file__, zookeeper_config_path='configs/zookeeper.xml')
-node = cluster.add_instance('node', with_zookeeper=True)
+node = cluster.add_instance('node', with_zookeeper=True, user_configs=["configs/users.xml"])
```
Comment (nikitamikhaylov) [inline tests/integration/test_reload_zookeeper/test.py:11]: Why do we need this change?

Comment (amosbird) [reply tests/integration/test_reload_zookeeper/test.py:11]: It seems unnecessary to me.

Comment (nikitamikhaylov) [reply tests/integration/test_reload_zookeeper/test.py:11]: Ok, thanks.

Comment (nikitamikhaylov) [reply tests/integration/test_reload_zookeeper/test.py:11]: I understood, we need it to have an exception from readonly table.

Comment (haoch) [reply tests/integration/test_reload_zookeeper/test.py:11]: It's used to enable the required configuration in `configs/users.xml`

**Pull request merged**